### PR TITLE
hard code comments into a JSON string and parsed to add to DOM

### PR DESCRIPTION
**Pull request by miahsanchez**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,34 @@
...
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    ArrayList<String> list = new ArrayList<String>();
```
Comment (jesseelizabeth) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: Use List as the declared type here - `private List<String> list = new ArrayList<>();`

When declaring types it's best to be as abstract as possible. This way you have the flexibility to use other list types down the line without breaking your code.

File: portfolio/pom.xml [hunk 2]
```diff
@@ -23,6 +23,11 @@
       <version>4.0.1</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
+      <artifactId>gson</artifactId>
+      <version>2.8.6</version>
+</dependency>
```
Comment (jesseelizabeth) [inline portfolio/pom.xml:30]: Fix indentation

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -26,3 +26,15 @@ function addRandomGreeting() {
...
 }
+
+/**
+ * Fetches comments from the servers and adds them to the DOM
+ */
+ function getComments(){
+    fetch('/data').then(response => response.json()).then((stats) => {
+        document.getElementById('comment1').innerText = stats[0];
+        document.getElementById('comment2').innerText = stats[1];
+        document.getElementById('comment3').innerText = stats[2];
```
Comment (jesseelizabeth) [inline portfolio/src/main/webapp/script.js:37]: This looks great, and definitely works. But what happens if you want to add another comment? Let's try to refactor this to be more dynamic. For example: we can have one 'comment-list' in our html, loop through our comments here, and append each to the comment-list.

Here's an example (don't worry about the delete functionality in this example yet): https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/examples/todo-list/src/main/webapp/script.js

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -19,14 +19,34 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:22]: Avoid importing entire packages -- there are some pitfalls to doing so (example: https://www.codebyamir.com/blog/pitfalls-java-import-wildcards), and also being explicit about which classes we use is better

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: Lets use more descriptive variable names -- "sampleContentList" or "sampleContents" something else that makes sense

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -19,14 +19,34 @@
...
+    String json = convertToJsonUsingGson(list);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(ArrayList<String> comments) {
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: From Jesse's comment about using declared types as abstract as possible - same goes for method arguments for the same reasons, so this should be List<String>

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -19,14 +19,34 @@
...
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    ArrayList<String> list = new ArrayList<String>();
+    list.add("My name is Miah");
+    list.add("Im from NYC");
+    list.add("I love coding!");
+
+    // Send the JSON as the response
+    String json = convertToJsonUsingGson(list);
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:38]: as a small nit I'd say we probably don't need this helper method - as we can just in one line say:
String jsonComment = new Gson().toJson(list);
and that one liner is pretty self explanatory

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -19,14 +19,34 @@
...
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: nit: the comment on "dependency added to pom.xml" isn't needed as that's a project setup impl detail unrelated to this class

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -19,15 +19,28 @@
...
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Miah!</h1>");
+
+    // TODO:  Make ArrayList more abstract aka make it a List.
+    ArrayList<String> sampleContents = new ArrayList<String>();
```
Comment (surbhimadan1995) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: you should be able to make this change in this PR itself, just commit your changes from this branch and they should be on this PR :) (keeping in line with "quick fix" style comments should be addressed before merging the PR)

**Pull request merged**